### ISSUE TITLE
Fake.Api.GitHub not compatible with newest Octokit 0.50?

### ISSUE THREAD

**Issue by MiloszKrajewski** [fsprojects/FAKE#2601]
### Description

After deleting `build.fsx.lock` (was so old it was not handling new MSBuild logs) and allowing FAKE to recreate it - uploading build artifacts to GitHub stopped working.

### Repro steps

1. Delete `build.fsx.lock`
2. Run fake and allow it to restore `build.fsx.lock`
3. Run task uplading to GitHub, ie:

```
GitHub.createClientWithToken token
|> GitHub.draftNewRelease user repository productVersion prerelease notes
|> GitHub.uploadFiles files
|> GitHub.publishDraft
|> Async.RunSynchronously
```

4. Observe:

```
Something failed, trying again: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Octokit.ReleaseAsset> Octokit.IReleasesClient.UploadAsset(Octokit.Release, Octokit.ReleaseAssetUpload)'.
   at Fake.Api.GitHub.uploadFile@212-7.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.AsyncPrimitives.unitAsync@577.Invoke(AsyncActivation`1 ctxt) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 577
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 105
Something failed, trying again: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Octokit.ReleaseAsset> Octokit.IReleasesClient.UploadAsset(Octokit.Release, Octokit.ReleaseAssetUpload)'.
   at Fake.Api.GitHub.uploadFile@212-7.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.AsyncPrimitives.unitAsync@577.Invoke(AsyncActivation`1 ctxt) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 577
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 105
Something failed, trying again: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Octokit.ReleaseAsset> Octokit.IReleasesClient.UploadAsset(Octokit.Release, Octokit.ReleaseAssetUpload)'.
   at Fake.Api.GitHub.uploadFile@212-7.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.AsyncPrimitives.unitAsync@577.Invoke(AsyncActivation`1 ctxt) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 577
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 105
Something failed, trying again: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Octokit.ReleaseAsset> Octokit.IReleasesClient.UploadAsset(Octokit.Release, Octokit.ReleaseAssetUpload)'.
   at Fake.Api.GitHub.uploadFile@212-7.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.AsyncPrimitives.unitAsync@577.Invoke(AsyncActivation`1 ctxt) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 577
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 105
Something failed, trying again: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Octokit.ReleaseAsset> Octokit.IReleasesClient.UploadAsset(Octokit.Release, Octokit.ReleaseAssetUpload)'.
   at Fake.Api.GitHub.uploadFile@212-7.Invoke(Unit unitVar)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 386
   at Microsoft.FSharp.Control.AsyncPrimitives.unitAsync@577.Invoke(AsyncActivation`1 ctxt) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 577
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 105
```

### Expected behavior

Uploading files to github.

### Actual behavior

Release is created but files are not updloaded.
Looks like `Octokit.IReleasesClient.UploadAsset` no longer exists.

### Known workarounds

Forcing specific version of Octokit: `nuget Octokit 0.48`

### Related information 

* Operating system: Windows 10
* .NET Runtime, CoreCLR or Mono Version: .NET 5
* Fake version: 5.20.4
* Paket version 6.0.3

### build.fsx (without forced Octokit version)

```
#r "paket:
    nuget Fake.Core.Target
    nuget Fake.Core.ReleaseNotes
    nuget Fake.IO.FileSystem
    nuget Fake.IO.Zip
    nuget Fake.Api.GitHub
    nuget Fake.DotNet.MSBuild
    nuget Fake.DotNet.Cli
    nuget Fake.DotNet.Testing.XUnit2
//"
```

**Comment by yazeedobaid** [comment]
Octokit has been updated to version 0.50 in FAKE and the UploadAsset API has been updated to match with Octokit's new API.
Can you please try one of the alpha releases and see if the issue still exists for you

Thanks

**Comment by yazeedobaid** [comment]
The fix for this issue is included in release [5.21.0](https://github.com/fsprojects/FAKE/releases/tag/5.21.0)

Thanks
